### PR TITLE
Use on_repair when repairing tube

### LINKED ISSUE #36: Hammer tube repair issue with TP tubes
TP tubes refusing to continue sending when broken and then repaired again.

Tubes should have optional repair method to allow overriding default minetest.swap_node repair behavior with additional stuff.
That makes fixing simple while keeping full compatibility and allowing any behavior node requires for repair.

Just call new `def.on_repair` or `def.pipeworks_on_repair` or whatever here if it is defined or use old behavior as fallback:
https://github.com/mt-mods/pipeworks/blob/7b15bdbd1f46f4a804dfb0c5a3570081a70b8db0/routing_tubes.lua#L86-L88

### PR DESCRIPTION
**Pull request by S-S-X**
Fixes #36 

I've not tested, not even launched mt. But GH web editor is way better nowdays 😄

### CODE REVIEW

**Comment by wsor4035** [comment]
mods should use the _ prefix per lua_api.txt

**Comment by OgelGames** [comment]
Yeah, no one uses the prefix :P I would suggest putting it inside the `tube` table though, to keep all the pipeworks stuff in one place.

Side note: `is_teleport_tube` is also defined outside of the `tube` table, but I noticed it's only used in one place:
https://github.com/mt-mods/pipeworks/blob/37eef736950f038e54fd74d11a5d905fb43abdad/teleport_tube.lua#L124
Might be good to change this to check the name instead, so that value can be removed. (in another PR)

File: teleport_tube.lua [hunk 1]
```diff
@@ -259,6 +259,18 @@ pipeworks.register_tube("pipeworks:teleport_tube", {
 		end,
 		on_destruct = function(pos)
 			remove_tube(pos)
+		end,
+		on_repair = function(pos, node)
```
[the remark this answers is not in the document]
Comment (S-S-X) [reply teleport_tube.lua]: 64fb579 moved to nodedef.tube.on_repair

**Comment by S-S-X** [comment]
Switched to nodedef.tube.on_repair, also added extra guard against possible unknown node crash (very unlikely but possible and cheap check).
I've still not tested or launched server with these changes.

**Comment by OgelGames** [comment]
> Not sure if swap_node can tolerate unknown...

Yeah it can't:
```
AsyncErr: Lua: Runtime error from mod 'pipeworks' in callback ScriptApiNode::node_on_punch(): "nothing:nothing" is not a registered node!
stack traceback:
	[C]: in function 'swap_node'
	C:\Minetest\client\bin\..\mods\pipeworks/routing_tubes.lua:87: in function <C:\Minetest\client\bin\..\mods\pipeworks/routing_tubes.lua:63>
```
Seems to work correctly otherwise, at least from my limited testing.

**Comment by S-S-X** [comment]
Now unknown node situation is checked for all paths instead of just for new functionality, situation is logged and also message sent to player telling that tube cannot be repaired.

**Review by OgelGames — APPROVED** [review]

**Pull request merged**